src/vm/types/layout.zig

```diff
@@ -153,6 +166,90 @@ pub const CairoLayout = struct {
         };
     }
 
+    /// Sets up the built-in runners for the layout.
+    ///
+    /// # Arguments
+    ///
+    /// - `self`: The layout structure.
+    /// - `allocator`: Allocator for memory management.
+    /// - `proof_mode`: Whether the program is in proof mode.
+    /// - `program_builtins`: The builtins used by the executed program.
+    ///
+    /// # Returns
+    ///
+    /// An array list of builtin runners.
+    pub fn setUpBuiltinRunners(
+        self: Self,
+        allocator: Allocator,
+        proof_mode: bool,
+        program_builtins: []const []const u8,
+    ) !ArrayList(BuiltinRunner) {
+        var builtin_runners = ArrayList(BuiltinRunner).init(allocator);
+
+        if (proof_mode) {
+            // TODO: initialize all the builtins in the layout.
+        }
+
+        const Case = enum {
+            output,
+            pedersen,
+            range_check,
+            ecdsa,
+            bitwise,
+            ec_op,
+            keccak,
+            poseidon,
+        };
+
+        for (program_builtins) |builtin| {
+            const case = std.meta.stringToEnum(Case, builtin) orelse return RunnerError.BuiltinNotInLayout;
+            switch (case) {
+                .output => if (self.builtins.output) {
+                    try builtin_runners.append(BuiltinRunner{ .Output = OutputBuiltinRunner.initDefault(allocator) });
+                },
+
+                // TODO: implement initDefault for the rest of the builtin runners.
+
+                .pedersen => if (self.builtins.pedersen != null) {
```

Comment:
there is a school of thought that says writing code in a 'return early' style with the nonhappy path is easier to read. i tend to agree. see: https://medium.com/swlh/return-early-pattern-3d18a41bba8

if we were to apply this principle here we could do something like:

```
fn isBuiltinAvailable(self: *Self, case: Builtin) bool {
    return switch (case) {
        .output => self.builtins.output != null,
        .pedersen => self.builtins.pedersen != null,
        .range_check => self.builtins.range_check != null,
        .ecdsa => self.builtins.ecdsa != null,
        .bitwise => self.builtins.bitwise != null,
        .ec_op => self.builtins.ec_op != null,
        .keccak => self.builtins.keccak != null,
        .poseidon => self.builtins.poseidon != null,
    };
}
```

then we can have early check if we are in the failing case with a return

`if (!self.isBuiltinAvailable(case)) return RunnerError.BuiltinNotInLayout;`

and another switch that handles constructing/appending the right builtin

lmk what you think

Reply:
Great suggestion, improved readability a ton